### ISSUE TITLE
Android - Connect Manually -The App can crash after clicking on the back button during manual Bank account flow

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#13857]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to the WS settings
2. Connect bank account --> Connect manually , enter Routing/Account numbers and open Company information dialog
3. Click few times on the back button until workspace menu is open again
4. Repeat 2-3 until crash is reproduced

## Expected Result:
The user should be able to use back button during manual bank account flow without any issues

## Actual Result:
The App can crash after clicking on the back button during manual Bank account flow

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?

- [ ] Android / native

**Version Number:** 1.2.44.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/209841711-2a2ebf1b-b7e8-4124-a797-9bafe439801c.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by puneetlath** [comment]
I was able to reproduce this. I got this error on the emulator. 

<img width="409" alt="image" src="https://user-images.githubusercontent.com/1371996/210452725-34b137ee-1411-4601-97f7-8b60f05bd8ee.png">

**Comment by allroundexperts** [comment]
### Proposal
The error occurs because the component un mounts before the timeout is executed here.
https://github.com/allroundexperts/Expensify/blob/d9ec05a2a26c42e865974322a2e9006b327f95d2/src/components/TextInput/BaseTextInput.js#L69

**Comment by Pujan92** [comment]
**Proposal**

As @allroundexperts pointed we can check the input nullable or if the input is available then only it should call the `focus` method.

```diff
diff --git a/src/components/TextInput/BaseTextInput.js b/src/components/TextInput/BaseTextInput.js
index 0fed14ea4..865054418 100644
--- a/src/components/TextInput/BaseTextInput.js
+++ b/src/components/TextInput/BaseTextInput.js
@@ -66,7 +66,7 @@ class BaseTextInput extends Component {
         }
 
         if (this.props.shouldDelayFocus) {
-            return setTimeout(() => this.input.focus(), CONST.ANIMATED_TRANSITION);
+            return setTimeout(() => this.input && this.input.focus(), CONST.ANIMATED_TRANSITION);
         }
         this.input.focus();
```

**Comment by syedsaroshfarrukhdot** [comment]
Proposal :-

This error occurs when you are trying to access a property of an object that is null or undefined. In the case of React Native, it is likely that you are trying to call the focus method on a TextInput component that has not yet been rendered or that has been unmounted.

Solution to this problem is to make sure that the component is only accessed after it has been rendered. You can do this by using an if statement to check if the component is null before trying to call the focus method.

``` diff 

diff --git a/src/components/TextInput/BaseTextInput.js b/src/components/TextInput/BaseTextInput.js
index 0fed14ea4..85da8cade 100644
--- a/src/components/TextInput/BaseTextInput.js
+++ b/src/components/TextInput/BaseTextInput.js
@@ -65,7 +65,7 @@ class BaseTextInput extends Component {
             return;
         }
 
-        if (this.props.shouldDelayFocus) {
+        if (this.props.shouldDelayFocus && this.textInput != null) {
             return setTimeout(() => this.input.focus(), CONST.ANIMATED_TRANSITION);
         }
         this.input.focus();

```

**Comment by Santhosh-Sellavel** [comment]
@puneetlath

@0xmiroslav proposal looks good to me!
👀 🎀 👀 
C+ Reviewed

Thanks!

**Comment by puneetlath** [comment]
I agree that canceling the timeout when unmounting is a good approach. Thanks everyone for your proposals. Let's go with @0xmiroslav for this one.

**Comment by allroundexperts** [comment]
@puneetlath @Santhosh-Sellavel Have you seen my proposal? I mentioned this solution at the very beginning.

**Comment by allroundexperts** [comment]
I think it was more than clear what I was trying to say here.

**Comment by puneetlath** [comment]
Hi @allroundexperts. I agree that your proposal wasn't as clear in what it was trying to do. Also, I didn't realize you had added a solution. When I first saw your comment, it only said the problem, without referencing a solution. I see now that it has been edited to include a solution. I think in this case, it was fair to give @0xmiroslav the job since their proposal was more clear. Looking forward to working with you in future jobs though!

**Comment by allroundexperts** [comment]
> Hi @allroundexperts. I agree that your proposal wasn't as clear in what it was trying to do. Also, I didn't realize you had added a solution. When I first saw your comment, it only said the problem, without referencing a solution. I see now that it has been edited to include a solution. I think in this case, it was fair to give @0xmiroslav the job since their proposal was more clear. Looking forward to working with you in future jobs though!

@puneetlath Please check the time when I edited this. It was at least 24 hours before @0xmiroslav posted his solution. I still think that my solution is clear enough for anyone with good React knowledge to understand. This is not the first time I'm contributing here and I have made several successful contributions in past as well. I believe this is unfair as in past several of my proposals were rejected because someone had already proposed it before.

**Comment by puneetlath** [comment]
Hey @allroundexperts I can understand your frustration, but both @Santhosh-Sellavel and I didn't find it clear enough to decide to go with it. There will be plenty more jobs to come.

**Comment by puneetlath** [comment]
@Santhosh-Sellavel @0xmiroslav can you please apply to the Upwork job? https://www.upwork.com/jobs/~01c5662eb8534df401

**Comment by puneetlath** [comment]
Paid @0xmiroslav with 50% bonus. Just waiting for @Santhosh-Sellavel to accept the offer. Then we'll be good to close.

**Comment by puneetlath** [comment]
All paid. Thanks everyone!
